test/jdk/sun/security/pkcs11/MessageDigest/ReinitDigest.java

```diff
@@ -94,4 +93,17 @@ private static void check(byte[] d1, byte[] d2) throws Exception {
             throw new RuntimeException("Digest mismatch");
         }
     }
+
+    private static void checkInstances(MessageDigest md1, MessageDigest md2)
+            throws Exception {
+        if (md1.equals(md2)) {
+            throw new RuntimeException("MD instances should be different");
+        }
+        if (!md1.getAlgorithm().equals(md2.getAlgorithm())) {
+            throw new RuntimeException("Algorithm name should equal");
+        }
+        if (md1.getProvider().equals(md2.getProvider())) {
```

Comment:
nit: calls getName() since the error message says that the comparison is based on provider name.

Reply:
Updated. It was meant to be provider name check. Thanks..